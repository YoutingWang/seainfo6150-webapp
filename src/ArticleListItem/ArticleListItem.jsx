import React from "react";

const ArticleListItem = (props) => {
  return (
    <article>
      <h1>{props.article.title}</h1>
      <time dateTime={props.article.timeStamp}>{props.article.displayDate}</time>
      <p>{props.article.shortText}</p>
    </article>
  );
};

export default ArticleListItem;