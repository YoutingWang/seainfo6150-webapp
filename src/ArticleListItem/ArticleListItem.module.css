.articleTitle {
    font-weight: bold;
    color: #ca0303;
    margin-top: 0.2em;
    text-decoration: underline;
}
.articleBody{
    margin-bottom: 20px; 
    background: linear-gradient(to right, #bb3b3b, white 20%);
    box-shadow: 4px 4px 7px;
    padding: 0 8px 8px 8px;
    border: 2px solid #b40303;
    border-radius: 12px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
}
.dateTime {
    font-weight: bold;
}
