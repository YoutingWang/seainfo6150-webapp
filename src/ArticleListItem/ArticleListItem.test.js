import React from "react";
import ArticleListItem from "./ArticleListItem";
import {
    fireEvent
} from "@testing-library/react";

describe("ArticleListItem tests", () => {
            it("renders correctly", () => {
                    const article = {
                        slug: "181122102925-statue-of-liberty-crown",
                        title: "The Statue of Liberty's torch heads to new museum",
                        shortText: "The Statue of Liberty's original torch is getting a new home",
                        displayDate: "November 22nd 2018, 7:12 am"
                    };
                    const {
                        container
                    } = render( < ArticleListItem article = {
                            article
                        }
                        />);
                        expect(container).toMatchSnapshot();
                    });
            });