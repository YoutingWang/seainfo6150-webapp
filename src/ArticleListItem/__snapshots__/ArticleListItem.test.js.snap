// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListItem tests renders correctly 1`] = `
<div>
  <article
    class="articleBody"
  >
    <h2
      class="articleTitle"
    >
      The Statue of Liberty's torch heads to new museum
    </h2>
    
    <button
      class="toggleButton"
    >
      Show more
    </button>
  </article>
</div>
`;
